sm/plugins/tedge_apt_plugin/src/main.rs

```diff
@@ -73,14 +65,41 @@ fn run(operation: PluginOp) -> Result<ExitStatus, InternalError> {
             status
         }
 
-        PluginOp::Install { module, version } => {
-            if let Some(version) = version {
-                run_cmd(
+        PluginOp::Install {
+            module,
+            version,
+            file_path,
+        } => {
+            match (&version, &file_path) {
+                (Some(version), Some(file_path)) => {
+                    let file_path = validate_package(
+                        file_path,
+                        &[
+                            &format!("Version: {}", &version),
+                            &format!("Package: {}", &module),
+                            "Debian package",
+                        ],
+                    )?;
+                    run_cmd("apt-get", &format!("install --quiet --yes {}", file_path))?
+                }
+
+                (Some(version), None) => run_cmd(
                     "apt-get",
                     &format!("install --quiet --yes {}={}", module, version),
-                )?
-            } else {
-                run_cmd("apt-get", &format!("install --quiet --yes {}", module))?
+                )?,
+
+                (None, Some(file_path)) => {
+                    let file_path = validate_package(
+                        file_path,
```

Comment:
It's not obvious as first read that the file_path is changed with a proper extension added.